iceball/src/lib.rs

```diff
@@ -1470,4 +1470,378 @@ pub enum X64Statement {
+    /// - PF(fault-code): For a page fault.
+    /// - AC(0): If alignment checking is enabled and an unaligned memory reference is made while the current privilege level is 3.
+    ///
+    /// ### Real-Address Mode Exceptions
+    /// - UD: If LOCK prefix is used.
+    /// - UD: If CPUID.(EAX=07H, ECX=0H):EBX.ADX[bit 19] = 0.
+    /// - SS(0): For an illegal address in the SS segment.
+    /// - GP(0): If any part of the operand lies outside the effective address space from 0 to FFFFH.
+    ///
+    /// ### Virtual-8086 Mode Exceptions
+    /// - UD: If LOCK prefix is used.
+    /// - UD: If CPUID.(EAX=07H, ECX=0H):EBX.ADX[bit 19] = 0.
+    /// - SS(0): For an illegal address in the SS segment.
+    /// - GP(0): If any part of the operand lies outside the effective address space from 0 to FFFFH.
+    /// - PF(fault-code): For a page fault.
+    /// - AC(0): If alignment checking is enabled and an unaligned memory reference is made while the current privilege level is 3.
+    ///
+    /// ### Compatibility Mode Exceptions
+    /// - UD: If LOCK prefix is used.
+    /// - UD: If CPUID.(EAX=07H, ECX=0H):EBX.ADX[bit 19] = 0.
+    /// - SS(0): For an illegal address in the SS segment.
+    /// - GP(0): For an illegal memory operand effective address in the CS, DS, ES, FS, or GS segments.
+    /// - GP(0): If the DS, ES, FS, or GS register is used to access memory and it contains a NULL segment selector.
+    /// - PF(fault-code): For a page fault.
+    /// - AC(0): If alignment checking is enabled and an unaligned memory reference is made while the current privilege level is 3.
+    ///
+    /// ### 64-Bit Mode Exceptions
+    /// - UD: If LOCK prefix is used.
+    /// - UD: If CPUID.(EAX=07H, ECX=0H):EBX.ADX[bit 19] = 0.
+    /// - SS(0): If a memory address referencing the SS segment is in a non-canonical form.
+    /// - GP(0): If the memory address is in a non-canonical form.
+    /// - PF(fault-code): For a page fault.
+    /// - AC(0): If alignment checking is enabled and an unaligned memory reference is made while the current privilege level is 3.
+    ///
+    /// ## Intel C/C++ Compiler Intrinsic Equivalent
+    /// unsigned char _addcarryx_u32 (unsigned char c_in, unsigned int src1, unsigned int src2, unsigned int *sum_out);
+    /// unsigned char _addcarryx_u64 (unsigned char c_in, unsigned __int64 src1, unsigned __int64 src2, unsigned __int64 *sum_out);
+    ///
+    /// ## Operation
+    /// ```ignore
+    /// IF OperandSize is 64-bit
+    ///     THEN OF:DEST[63:0] := DEST[63:0] + SRC[63:0] + OF;
+    ///     ELSE OF:DEST[31:0] := DEST[31:0] + SRC[31:0] + OF;
+    /// FI;
+    /// ```
+    Adox,
+    /// # aesdec
+    ///
+    /// Perform One Round of an AES Decryption Flow
+    ///
+    /// - aesdec xmm1, xmm2/m128 - Perform one round of an AES decryption flow, using the Equivalent Inverse Cipher, using one 128-bit data (state) from xmm1 with one 128-bit round key from xmm2/m128.
```

Comment:
AES encryption has some version about various key size, aes128, aes196, aes256, aes512, so these instructions are intended perform matching key-size AES encryption (but not 196, due to xmm ymm zmm can't be represented to fit 196 bit)